File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/player/action/PlayerEncounterAction.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.action;

import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Player;
import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Ship;

/**
 * Represents an action the player can do during an encounter.
 */
public interface PlayerEncounterAction {

    /**
     * Executes an action based on the player's selection. Depends on the
     * action implementing this interface.
     *
     * @param player the player
     * @param otherShip the other ship
     * @param es the state of the encounter (over or not over)
     * @return a string detailing what is happening
     */
    public String doAction(Player player, Ship otherShip, EncounterState es);
}

File: ios/SpaceTrader/SpaceTrader/src/entity/player/action/PlayerEncounterAction.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

protocol PlayerEncounterAction {
    
    func doAction(player : Player, otherShip : Ship, es : EncounterState) -> String
    
}
